GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs

```diff
@@ -0,0 +1,375 @@
+        {
+            return ConsoleHelper.ShowStatusWhileRunning(
+                method,
+                message,
+                this.output,
+                this.output == Console.Out && !GVFSPlatform.Instance.IsConsoleOutputRedirectedToFile(),
+                null);
+        }
+
+        private bool TryRunUpgradeInstall(out Version newVersion, out string error)
+        {
+            newVersion = null;
+
+            Version newGVFSVersion = null;
+            GitVersion newGitVersion = null;
+            string errorMessage = null;
+            if (!this.LaunchInsideSpinner(
+                () =>
+                {
+                    if (!this.TryCheckIfUpgradeAvailable(out newGVFSVersion, out errorMessage) ||
+                        !this.TryGetNewGitVersion(out newGitVersion, out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    this.LogInstalledVersionInfo();
+                    this.LogVersionInfo(newGVFSVersion, newGitVersion, "Available Version");
+
+                    if (!this.preRunChecker.TryRunPreUpgradeChecks(newGitVersion, out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    if (!this.TryDownloadUpgrade(newGVFSVersion, out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    return true;
+                },
+                "Downloading"))
+            {
+                error = errorMessage;
+                return false;
+            }
+
+            if (!this.LaunchInsideSpinner(
+                () =>
+                {
+                    if (!this.preRunChecker.TryUnmountAllGVFSRepos(out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    this.remount = true;
+
+                    if (!this.TryInstallGitUpgrade(newGitVersion, out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    return true;
+                },
+                "Installing Git"))
+            {
+                error = errorMessage;
+                return false;
+            }
+
+            if (!this.LaunchInsideSpinner(
+                () =>
+                {
+                    if (!this.TryInstallGVFSUpgrade(newGVFSVersion, out errorMessage))
+                    {
+                        return false;
+                    }
+
+                    return true;
+                },
+                "Installing GVFS"))
+            {
+                error = errorMessage;
+                return false;
+            }
+
+            this.LogVersionInfo(newGVFSVersion, newGitVersion, "Newly Installed Version");
+
+            newVersion = newGVFSVersion;
+            error = null;
+            return true;
+        }
+        
+        private bool TryRunCleanUp(out string error)

```

Comment:
This says "RunCleanup", but it's mounting. I would split this up into two methods.

See my earlier comment where I was complaining about why you're showing cleanup messages to users. Users won't care if some temp file failed to be deleted. They most definitely will care if one of their repos failed to mount.